src/base/bittorrent/peerinfo.cpp

```diff
@@ -230,24 +230,14 @@ void PeerInfo::calcRelevance(const Torrent *torrent)
 {
     const QBitArray allPieces = torrent->pieces();
     const QBitArray peerPieces = pieces();
+    Q_ASSERT(allPieces.size() == peerPieces.size());
```

Comment:
Is assertion required? AFAIK, QBitArray can apply `operator&` to arrays of different sizes.

Reply:
>Is assertion required? 

Not really. But I wanted to make sure the bit arrays are truly equivalent in size, otherwise the bit-wise operations won't make sense (logically).

Reply:
(It's just a guess... it can have a reduced size, not including trailing zeros.) 

Anyway, you should be sure that they can have different sizes only in case of an error somewhere in the code. Otherwise you should remove the assertion.

Reply:
>it can have a reduced size, not including trailing zeros

I won't be expecting it leaving out trailing zeros, the meaning would be different.

Removed anyway.